programs/exchange_api/src/exchange_processor.rs

```diff
@@ -40,8 +48,12 @@ impl ExchangeProcessor {
         }
     }
 
-    fn deserialize_trade(data: &[u8]) -> Result<(TradeOrderInfo), InstructionError> {
-        let state: ExchangeState = bincode::deserialize(data).map_err(Self::map_to_invalid_arg)?;
+    fn deserialize_trade(
+        keyed_accounts: &mut [KeyedAccount],
+        index: usize,
+    ) -> Result<(TradeOrderInfo), InstructionError> {
+        let state: ExchangeState = bincode::deserialize(&keyed_accounts[index].account.data[..])
```

Comment:
https://github.com/solana-labs/solana/blob/867f6f107b3c11090bc39de0bc80e5a424c8dd06/programs/stake_api/src/stake_state.rs#L42-L60

Reply:
also..  why is the return value `Result<(TradeOrderInfo), ..>`  instead of `Result<TradeOrderInfo, ..>`